bundles/org.openhab.binding.withings/src/main/java/org/openhab/binding/withings/internal/handler/AbstractWithingsThingHandler.java

```diff
@@ -0,0 +1,128 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Sven Strohschein - Initial contribution
+ */
+@NonNullByDefault
+public abstract class AbstractWithingsThingHandler extends BaseThingHandler implements WithingsThingHandler {
+
+    private static final DateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");
+
+    private final Logger logger = LoggerFactory.getLogger(AbstractWithingsThingHandler.class);
+
+    public AbstractWithingsThingHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        Optional<Bridge> bridgeOptional = getBridgeOptional();
+        if (bridgeOptional.isPresent()) {
+            updateStatusDependingFromBridgeStatus(bridgeOptional.get().getStatus());
+
+            registerAtBridge();
+        } else {
+            logger.error("The bridge couldn't get found!");
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (RefreshType.REFRESH.equals(command)) {
+            getBridgeHandlerOptional().ifPresent(bridge -> bridge.notifyThingHandler(this));
+        }
+    }
+
+    @Override
+    public final void updateData(WithingsDataModel model) {
+        boolean isUpdated = updateThingData(model);
+
+        if (isUpdated) {
+            updateStatus(ThingStatus.ONLINE);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.GONE);
+        }
+    }
+
+    @Override
+    protected void updateStatus(ThingStatus status, ThingStatusDetail statusDetail, @Nullable String description) {
+        super.updateStatus(status, statusDetail, description);
+    }
+
+    @Override
+    protected void updateStatus(ThingStatus status, ThingStatusDetail statusDetail) {
+        super.updateStatus(status, statusDetail);
+    }
+
+    @Override
+    protected void updateStatus(ThingStatus status) {
+        super.updateStatus(status);
+    }
+
+    protected abstract boolean updateThingData(WithingsDataModel model);
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        updateStatusDependingFromBridgeStatus(bridgeStatusInfo.getStatus());
+    }
+
+    protected DateTimeType createDateTimeType(Date date) {
+        return new DateTimeType(DATE_FORMAT.format(date));
+    }
+
+    private void registerAtBridge() {
+        Optional<WithingsBridgeHandler> bridgeHandlerOptional = getBridgeHandlerOptional();
+        if (bridgeHandlerOptional.isPresent()) {
+            bridgeHandlerOptional.get().registerThingHandler(this);
+        } else {
+            logger.error("The bridge handler couldn't get found!");
+        }
+    }
+
+    private void updateStatusDependingFromBridgeStatus(ThingStatus bridgeStatus) {
+        if (!ThingStatus.ONLINE.equals(bridgeStatus) && !ThingStatus.INITIALIZING.equals(bridgeStatus)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+        }
+    }
```

Comment:
That didn't worked unfortunately. Therefore I built that code and saw similar code in other bindings. Just search for the BRIDGE_OFFLINE constant for example. Maybe we should open an issue for OpenHAB.

Reply:
What was the error you observed without this code?

Reply:
That the Things were still displayed as ONLINE even when the bridge got OFFLINE and couldn't access the Withings API anymore (for example when I enter an invalid auth-code). I could try to remove the logic and test again sometime (but I expect that it will show the problems again).

Reply:
Yes, please try again. Maybe the web session has become invalid and the UI didn't update anymore?